mbed_devices/_internal/windows/windows_identifier.py

```diff
@@ -0,0 +1,68 @@
+"""Defines a Windows identifier."""
+from typing import NamedTuple, Any, Optional
+
+from mbed_devices._internal.windows.component_descriptor_utils import is_undefined_value
+
+INSTANCE_ID_CHARACTER = "&"
+
+
+class WindowsUID(NamedTuple):
+    """Definition of a Windows Universal ID."""
+
+    UID: str  # Processed string which may be the closest to the Universal ID.
+    raw_uid: str  # The String from which the UID was derived. Raw form as it can be a
+    # combination of other elements e.g. 8&3432E179&0&000
+    serial_number: Optional[str]  # a serial number
+
+    def __eq__(self, other: Any) -> bool:
+        """Defines the equality checker."""
+        if not other or not isinstance(other, WindowsUID):
+            return False
+        if (
+            self.UID == other.UID
+            or self.raw_uid == other.raw_uid
+            or self.serial_number == other.serial_number
+            or self.serial_number == other.UID
+            or self.UID == other.serial_number
+        ):
```

Comment:
The logic of this `__eq__` method is hard to follow. I think it needs a more descriptive docstring.

Reply:
28301b80044ce0a6df8120eed868e2979d6718dd